### DISCUSSION TITLE
Make it Function better without Javascript

### DISCUSSION THREAD

**Discussion by jermanuts** [withastro/starlight#1469] [Feature Requests]
### What version of `starlight` are you using?

0.17.3

### What is your idea?

It would be great if the language menu worked without Javascript.

For example, https://github.com/ultrabug/mkdocs-static-i18n.

privacyguides.org uses it and works great.

and the hamburger menu doesn't work on mobile when JS is disabled. It works out the box with mkdocs material.

### Why is this feature necessary?

Many privacy conscious users disable Javascript when browsing, having starlight work for these users without JS puts it on par with existing docs generators out there that also work great without JS.

### Do you have examples of this feature in other projects?

- https://github.com/ultrabug/mkdocs-static-i18n.
- https://squidfunk.github.io/mkdocs-material/

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Thanks for the suggestion! We’d definitely like things to work as much as possible without JavaScript.

However, highest importance will be to ensure that everything is fully accessible and works for screen readers and other assistive technology, which is notably not the case in the MkDocs Material theme. So if we can solve things in ways that are accessible AND work without JavaScript, I’ll be happy, but there may still be some patterns that require a little JS to be fully accessible.

**Reply by squidfunk** [reply]
Author of Material for MkDocs here. I'd be very interested to learn what parts are not accessible. As you may know, making sure that everything is accessible is an uphill battle and we're always interested to learn where we can improve, so if you can elaborate a bit on your claim that Material for MkDocs is not accessible, that would be great and fair!

**Reply by delucis** [reply]
Hey @squidfunk! I think the thing that really stood out to me was that I wasn’t able to reach the small viewport menu at all via the keyboard (i.e. tabbing through the page never seemed to reach the hamburger button). There are also several instances where things lack focus styling — i.e. if tabbed to with the keyboard there is no visual indication that they are focused. I didn’t do a thorough audit, but that’s what I recall from my quick tests — hope it helps!

**Reply by squidfunk** [reply]
Thanks, that's great input! We're constantly looking to improve A11Y and indeed, the mobile navigation is tricky to get right via keyboard so that it works without JavaScript. We're going to improve accessibility for it next time work on it ☺️

**Comment by HiDeoo** [comment]
As mentioned by Chris, if we can solve things in ways that are accessible AND work without JavaScript, that is definitely the best outcome. However, making things work without JavaScript no matter what is not a goal for Starlight if it comes at the cost of accessibility.

Considering this, I think we can close this discussion for now but we would definitely be happy to discuss targetted changes removing some JavaScript parts if they can be made fully accessible without it.
